src/com/ms/silverking/cloud/dht/client/gen/CPPWrapperGenerator.java

```diff
@@ -5,252 +5,255 @@
+    p("\tskw_jenv->CallXXXXMethod(instance, _class, %s, %s);", getMethodFieldName(m), getMethodParameters(m));
+  }
+
+  private void generateFields(Class c, Mode mode) {
+    for (Method m : c.getMethods()) {
+      if (m.getDeclaringClass().equals(c)) {
+        generateFieldsForMethod(c, m, mode);
+      }
     }
-    
-    private String typeToField(Class t) {
-        String    n;
-        
-        n = t.getName();
-        n = n.replace('[', '_');
-        return n;
+  }
+
+  private void generateFieldsForMethod(Class c, Method m, Mode mode) {
+    if (mode == Mode.Definition) {
+      p("\tjmethodID\t%s;", getMethodFieldName(m));
+    } else {
     }
-    
-    private String getMethodFieldName(Method m) {
-        StringBuffer    sb;
-        boolean            isFirst;
-        
-        sb = new StringBuffer();
-        sb.append("id_"+ m.getName());
-        isFirst = true;
-        for (Parameter p : m.getParameters()) {
-            if (!isFirst) {
-                sb.append("__");
-            }
-            sb.append(typeToField(p.getType()));
-            isFirst = false;
-        }
-        return sb.toString();
+  }
+
+  private String typeToField(Class t) {
+    String n;
+
+    n = t.getName();
+    n = n.replace('[', '_');
+    return n;
+  }
+
+  private String getMethodFieldName(Method m) {
+    StringBuffer sb;
+    boolean isFirst;
+
+    sb = new StringBuffer();
+    sb.append("id_" + m.getName());
+    isFirst = true;
+    for (Parameter p : m.getParameters()) {
+      if (!isFirst) {
+        sb.append("__");
+      }
+      sb.append(typeToField(p.getType()));
+      isFirst = false;
     }
-    
-    private String getMethodSignature(Method m) {
-        StringBuffer    sb;
-        boolean            isFirst;
-        
-        sb = new StringBuffer();
-        isFirst = true;
-        for (Parameter p : m.getParameters()) {
-            if (!isFirst) {
-                sb.append(";");
-            }
-            sb.append(p.getType().getName());
-            isFirst = false;
-        }
-        return sb.toString();
+    return sb.toString();
+  }
+
+  private String getMethodSignature(Method m) {
+    StringBuffer sb;
+    boolean isFirst;
+
+    sb = new StringBuffer();
+    isFirst = true;
+    for (Parameter p : m.getParameters()) {
+      if (!isFirst) {
+        sb.append(";");
+      }
+      sb.append(p.getType().getName());
+      isFirst = false;
     }
-    
-    private String getMethodSignature(Method m, Mode mode) {
-        StringBuffer    sb;
-        boolean            isFirst;
-        
-        sb = new StringBuffer();
-        isFirst = true;
-        for (Parameter p : m.getParameters()) {
-            if (!isFirst) {
-                sb.append(", ");
-            }
-            sb.append(p.getType().getName());
-            sb.append(' ');
-            sb.append(p.getName());
-            isFirst = false;
-        }
-        return sb.toString();
+    return sb.toString();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 180:189, 214:223.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.